python/ray/_raylet.pyx

```diff
@@ -708,289 +713,302 @@ cdef execute_task(
-    def exit_current_actor_if_asyncio():
-        if core_worker.current_actor_is_asyncio():
-            error = SystemExit(0)
-            error.is_ray_terminate = True
-            error.ray_terminate_msg = "exit_actor() is called."
-            raise error
-
-    function_descriptor = CFunctionDescriptorToPython(
-        ray_function.GetFunctionDescriptor())
-
-    if <int>task_type == <int>TASK_TYPE_ACTOR_CREATION_TASK:
-        actor_class = manager.load_actor_class(job_id, function_descriptor)
-        actor_id = core_worker.get_actor_id()
-        actor = actor_class.__new__(actor_class)
-        worker.actors[actor_id] = actor
-        # Record the actor class via :actor_name: magic token in the log.
-        #
-        # (Phase 1): this covers code run before __init__ finishes.
-        # We need to handle this separately because `__repr__` may not be
-        # runnable until after `__init__` (e.g., if it accesses fields
-        # defined in the constructor).
-        actor_magic_token = "{}{}\n".format(
-            ray_constants.LOG_PREFIX_ACTOR_NAME, actor_class.__name__)
-        # Flush to both .out and .err
-        print(actor_magic_token, end="")
-        print(actor_magic_token, file=sys.stderr, end="")
-
-        # Initial eventloops for asyncio for this actor.
-        if core_worker.current_actor_is_asyncio():
-            core_worker.initialize_eventloops_for_actor_concurrency_group(
-                c_defined_concurrency_groups)
-
-    execution_info = execution_infos.get(function_descriptor)
-    if not execution_info:
-        execution_info = manager.get_execution_info(
-            job_id, function_descriptor)
-        execution_infos[function_descriptor] = execution_info
-
-    function_name = execution_info.function_name
-    extra_data = (b'{"name": ' + function_name.encode("ascii") +
-                  b' "task_id": ' + task_id.hex().encode("ascii") + b'}')
-
-    task_name = name.decode("utf-8")
-    name_of_concurrency_group_to_execute = \
-        c_name_of_concurrency_group_to_execute.decode("ascii")
-    title = f"ray::{task_name}()"
-
-    if <int>task_type == <int>TASK_TYPE_NORMAL_TASK:
-        next_title = "ray::IDLE"
-        function_executor = execution_info.function
-        # Record the task name via :task_name: magic token in the log file.
-        # This is used for the prefix in driver logs `(task_name pid=123) ...`
-        task_name_magic_token = "{}{}\n".format(
-            ray_constants.LOG_PREFIX_TASK_NAME, task_name.replace("()", ""))
-        # Print on both .out and .err
-        print(task_name_magic_token, end="")
-        print(task_name_magic_token, file=sys.stderr, end="")
-    else:
-        actor = worker.actors[core_worker.get_actor_id()]
-        class_name = actor.__class__.__name__
-        next_title = f"ray::{class_name}"
-
-        def function_executor(*arguments, **kwarguments):
-            function = execution_info.function
-
+    try:
```

Comment:
Could we separate out the try block into a separate function? i think it will be a bit more readable. Also, I think we will want to move the setting of `current_task_id` into this try block to prevent race conditions.

Reply:
Yep agreed, I was going for minimized changes in structures so the diff wouldn't be controversial and to lessen the risk of breaking weird invariants, but the change in indentation made the diff ugly anyway. 😛 

Btw, if we're separating it out into a separate function, we'll need to make sure that exceptions propagate properly, e.g. with the `except *` notation in the function signature. https://cython.readthedocs.io/en/stable/src/userguide/language_basics.html#error-return-values

Reply:
Got it, thanks for the tip! I'll give this a shot.